lighthouse-cli/index.js

```diff
@@ -140,35 +146,71 @@ if (cli.verbose) {
   flags.logLevel = 'error';
 }
 
-function runLighthouse(addresses) {
+const cleanups = [];
+process.on('SIGINT', () => {
+  cleanups.forEach(c => c());
+  process.exit(130);
+});
+
+function launchChromeAndRun() {
+  const launcher = new Launcher({
+    head: !cli.selectChrome,
+  });
+
+  cleanups.push(() => launcher.kill());
+
+  return launcher
+    .connect()
+    .catch(() => {
+      if (cli.skipAutolaunch) {
+        showConnectionError();
+      }
+      console.log('Launching Chrome...');
+      return launcher.run();
+    })
+    .then(() => lighthouseRun(urls))
+    .then(() => launcher.kill());
+}
+
+function lighthouseRun(addresses) {
   // Process URLs once at a time
   const address = addresses.shift();
   if (!address) {
     return;
   }
 
-  lighthouse(address, flags, config)
+  return lighthouse(address, flags, config)
     .then(results => Printer.write(results, outputMode, outputPath))
     .then(results => {
       if (outputMode !== 'html') {
         const filename = './' + assetSaver.getFilenamePrefix({url: address}) + '.html';
         Printer.write(results, 'html', filename);
       }
-      runLighthouse(addresses);
-      return;
+
+      return lighthouseRun(addresses);
     })
     .catch(err => {
       if (err.code === 'ECONNREFUSED') {
-        console.error('Unable to connect to Chrome.');
-        console.error('Please run Chrome w/ debugging port 9222 open:');
-        console.error('    npm explore -g lighthouse -- npm run chrome');
+        showConnectionError();
       } else {
-        console.error('Runtime error encountered:', err);
-        console.error(err.stack);
+        showRuntimeError(err);
       }
-      process.exit(1);
     });
 }
 
+function showConnectionError() {
+  console.error(
+    'Unable to connect to Chrome. Please run Chrome w/ debugging port 9222 open:'
+  );
+  console.error('    npm explore -g lighthouse -- npm run chrome');
```

Comment:
do we want to update this since the default way will now just let lighthouse handle launching chrome?